### DISCUSSION TITLE
Unable to deploy Vue Web to github.

### DISCUSSION THREAD

**Discussion by heash0000** [community/community#134128] [Programming Help]
### Body

I want to create and distribute the web using vue3 + pinia + typescript. 
The web operates normally when you do npm run serve locally.
However, I tried to build npm run and distribute using gh-pages but failed. 
What part of it did you miss?

It's my git project.
https://github.com/heash0000/ff14_collect
https://heash0000.github.io/ff14_collect/

**vue.config.js**
```javascript
const { defineConfig } = require('@vue/cli-service')
module.exports = defineConfig({
  transpileDependencies: true,
  publicPath: process.env.NODE_ENV === 'production'
    ? '/ff14-collect/'
    : '/'
})
```

**github setting page**
![image](https://github.com/user-attachments/assets/e653a8f3-3314-469a-aec4-bf2d1abe34d9)

error message
![image](https://github.com/user-attachments/assets/7b8980a2-cb8e-4498-8913-e2a1fa9ee253)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by marioprieta** [comment]
It looks like you're encountering 404 errors because your assets aren't loading correctly on GitHub Pages. Maybe you can try:

1. Check publicPath:

Ensure your vue.config.js is set correctly. It should look like this:

```
module.exports = {
  publicPath: process.env.NODE_ENV === 'production'
    ? '/ff14_collect/'
    : '/'
}
```

This ensures that paths are correctly prefixed with /ff14_collect/ in production.

2. Disable Jekyll:

GitHub Pages uses Jekyll by default, which might ignore files starting with _. Add a .nojekyll file to the root of your gh-pages branch to prevent this.


3. Rebuild and Redeploy:
Run npm run build to generate the production files.
Use a deployment tool like gh-pages to push the contents of the dist folder to the gh-pages branch.


4. Verify Asset Paths:
Ensure all asset paths in your HTML and JavaScript are relative or use the base path set by publicPath.


5. Use Hash Mode for Routing:
If you’re using Vue Router, switch to hash mode, as GitHub Pages doesn’t support HTML5 history mode without server-side configuration.

**Comment by aadityaa29** [comment]
Base URL Configuration:

In your vue.config.js file, set the publicPath property to match your GitHub Pages URL:
javascript
Copy code
module.exports = {
  publicPath: '/ff14_collect/', // Use your repository name
}
This ensures that the built assets are referenced correctly when hosted on GitHub Pages.
Building the Project:

Make sure to run npm run build to create the production-ready files in the dist folder.
Deploying the Build:

If you're using a script to deploy (like gh-pages), ensure you have the necessary deployment command set up in your package.json:
json
Copy code
"scripts": {
  "deploy": "gh-pages -d dist"
}
You can then run npm run deploy after building your project to push the dist folder to the gh-pages branch.
gh-pages Installation:

If you haven’t already, install the gh-pages package:
bash
Copy code
npm install gh-pages --save-dev
Check for Errors:

Review the console output when running npm run deploy for any errors. This can give you clues about what might be going wrong.
Browser Caching:

After deploying, if you don't see changes, make sure to clear your browser cache or try accessing the site in an incognito window.
Ensure Correct Branch:

Confirm that your gh-pages branch is being updated correctly. You can check this by navigating to the branch in your GitHub repository.
Check Console for Errors:

When accessing your site, open the developer console in your browser (usually F12) and check for any errors that might indicate issues with loading assets or running scripts.

**Reply by Sanidhyafeaturist** [reply]
To successfully deploy your Vue 3 project using GitHub Pages, ensure your vue.config.js is configured correctly for the publicPath, as you've done. Also, check that your build command (npm run build) creates the output in the dist directory and that you’re pushing the contents of dist to the gh-pages branch. You can automate this process using the gh-pages npm package by running npm run build followed by gh-pages -d dist. Make sure the repository settings are correct, and your GitHub Pages is set to serve from the gh-pages branch.
